src/web_peer_connection.cpp

```diff
@@ -789,9 +789,12 @@ void web_peer_connection::on_receive(error_code const& error
 			// if the status code is not one of the accepted ones, abort
 			if (!is_ok_status(m_parser.status_code()))
 			{
-				file_request_t const& file_req = m_file_requests.front();
-				m_web->have_files.resize(t->torrent_file().num_files(), true);
-				m_web->have_files.clear_bit(file_req.file_index);
+				if (!m_file_requests.empty())
+				{
+					file_request_t const& file_req = m_file_requests.front();
+					m_web->have_files.resize(t->torrent_file().num_files(), true);
```

Comment:
this resize() doesn't look right. it marks all files with lower indices as existing on this web seed..